test/ibmq/test_ibmq_job.py

```diff
@@ -385,41 +383,29 @@ def test_double_submit_fails(self, qe_token, qe_url):
         with self.assertRaises(JobError):
             job.submit()
 
+    @requires_qe_access
+    def test_error_message_qasm(self, qe_token, qe_url):
+        """Test retrieving job error messages including QASM status(es)."""
+        IBMQ.enable_account(qe_token, qe_url)
+        backend = IBMQ.get_backend('ibmq_qasm_simulator')
 
-@unittest.skip('Temporarily disabled, see #1162')
-class TestQobjBasedIBMQJob(JobTestCase):
-    """Test jobs supporting Qobj."""
-
-    def setUp(self):
-        super().setUp()
-        self._testing_device = os.getenv('IBMQ_QOBJ_DEVICE', None)
-        self._qe_token = os.getenv('IBMQ_TOKEN', None)
-        self._qe_url = os.getenv('IBMQ_QOBJ_URL')
-        if not self._testing_device or not self._qe_token or not self._qe_url:
-            self.skipTest('No credentials or testing device available for '
-                          'testing Qobj capabilities.')
+        qr = QuantumRegister(50)
+        cr = ClassicalRegister(2)
+        qc = QuantumCircuit(qr, cr)
+        qc.cx(qr[0], qr[1])
+        qc_new = transpile(qc, backend)
 
-        IBMQ.enable_account(self._qe_token, self._qe_url)
-        self._backend = IBMQ.get_backend(self._testing_device)
+        qobj = assemble(qc_new, shots=1000)
+        qobj.experiments[0].instructions[0].name = 'test_name'
 
-        self._qc = _bell_circuit()
+        job_sim = backend.run(qobj)
+        with self.assertRaises(JobError):
+            job_sim.result()
 
-    def test_qobj_enabled_job(self):
-        """Job should be an instance of IBMQJob."""
-        qobj = assemble(
-            transpile(self._qc, backend=self._backend), backend=self._backend)
-        job = self._backend.run(qobj)
-        self.assertIsInstance(job, IBMQJob)
-
-    def test_qobj_enabled_result(self):
-        """Jobs can be retrieved."""
-        qobj = assemble(
-            transpile(self._qc, backend=self._backend), backend=self._backend)
-        job = self._backend.run(qobj)
         try:
-            job.result()
-        except JobError as err:
-            self.fail(err)
+            job_sim.error_message()
```

Comment:
Instead of "trying and failing", can you use an assert:

```
something = job_sim.error_message()
self.assertIn('some string that is in the error message', something)
```